index.js

```diff
@@ -394,20 +428,14 @@ Color.prototype.setValues = function (space, vals) {
 
 	// cap values of the space prior converting all values
 	for (i = 0; i < space.length; i++) {
-		capped = Math.max(0, Math.min(maxes[space][i], this.values[space][i]));
-		this.values[space][i] = Math.round(capped);
+		capped = Math.max(0, Math.min(maxes[space][i], values[space][i]));
+		values[space][i] = Math.round(capped);
 	}
 
 	// convert to all the other color spaces
 	for (var sname in spaces) {
 		if (sname !== space) {
-			this.values[sname] = convert[space][sname](this.values[space]);
-		}
-
-		// cap values
-		for (i = 0; i < sname.length; i++) {
```

Comment:
The conversion is required because the whole lib assumes that all spaces are up-to-date anytime.